app/questionnaire/navigator.py

```diff
@@ -96,13 +97,13 @@ class Navigator:
     PRECEEDING_INTERSTITIAL_PATH = ['introduction']
     CLOSING_INTERSTITIAL_PATH = ['summary', 'thank-you']
 
-    def __init__(self, survey_json, metadata=None, answer_store=None):
+    def __init__(self, survey_json, metadata=None, answer_store=None, group_id=None, group_instance=0):
```

Comment:
I am not sure how to make this work, isn't the generation of the location path fundamental to init of the navigator? 
The __init__ before this change set the block_id,  group_id, group_instance, first_block_id, last_group_id and location_path, it just didn't pass anything in the __init__ as it always assumed it would be the first group/instance. I can no longer default to the first group or instance as the first time it linearly finds a repeating rule with no answer it stops  and therefore group navigation becomes impossible and will always go back to that first repeating rule.
Happy for anyone to pick up this branch, if they know how to make this work differently